pkg/qrcode/qrcode.go

```diff
@@ -59,3 +60,41 @@ func Parse(data []byte) (Data, error) {
 	}
 	return nil, errFormat
 }
+
+// EndDeviceFormat is a end device QR code format.
+type EndDeviceFormat interface {
+	Format() *ttnpb.QRCodeFormat
+	New() EndDeviceData
+}
+
+var (
+	endDeviceFormats   = map[string]EndDeviceFormat{}
+	endDeviceFormatsMu sync.RWMutex
```

Comment:
Are there any plans for dynamic registration of formats ? If they are only registered at `init()` I don't see why the mutex should be used.

Reply:
We don't really need the mutex indeed. It's doesn't hurt though. I didn't want to expose the `map[string]EndDeviceFormat{}` so it could be changed by anyone. And while I was at it, I protected it.